### PR TITLE
Add tests to execution plan

### PR DESCRIPTION
**Pull request by mnishamk-freiheit**
- Remove test workflow and move tests to existing workflow
- Helm test is moved to lint and test helm charts flow
- pkg and pkg/api are added are test-dependancies in frontend and cd-service (since they dont have independant buildfiles)
- cd-service and frontend-service tests are added as part of build-pr and build-main command

story: [SRX-57TKD2](https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/-M37cZx7XYoOvQ5Mmcir/STORY/-N3djfe-fMevnyrNqHzj/)

### CODE REVIEW

File: infrastructure/scripts/check-license/Buildfile [hunk 1]
```diff
@@ -4,3 +4,6 @@ metadata:
 spec:
   dependsOn:
   - ../../..
+  - ../../docker/builder
+  - ../../../services/cd-service
```
Comment (hannesg) [inline infrastructure/scripts/check-license/Buildfile:8]: Shouldn't that be redundant to `../../..`

File: Makefile.variables [hunk 2]
```diff
@@ -17,4 +17,4 @@
 VERSION?=$(shell git describe --always --long --tags || echo 0.0.1)
 IMAGE_SERVER?=europe-west3-docker.pkg.dev
 IMAGE_REGISTRY?=$(IMAGE_SERVER)/fdc-public-docker-registry/kuberpult
-KUBERPULT_BUILDER?=europe-west3-docker.pkg.dev/fdc-public-docker-registry/kuberpult/kuberpult-builder:0.0.1
+KUBERPULT_BUILDER?=europe-west3-docker.pkg.dev/fdc-public-docker-registry/kuberpult/kuberpult-builder:0.4.21-14-g146b8ea
```
Comment (medo-freiheit) [inline Makefile.variables:20]: Are we setting this manually?

File: charts/kuberpult/Makefile [hunk 3]
```diff
@@ -52,10 +52,14 @@ else
...
 endif
 
-test: ct-test $(TGZ_FILE)
+test-raw: $(TGZ_FILE)
 	helm lint $(TGZ_FILE) --set git.url=test --set ingress.domainName=kuberpult.example.com
 	helm lint $(TGZ_FILE) --set git.url=test --set ingress.domainName=kuberpult.example.com --set ingress.exposeReleaseEndpoint=true
 
+test-ci: test-raw
+
+test: ct-test test-raw
```
Comment (medo-freiheit) [inline charts/kuberpult/Makefile:61]: shouldn't be:
```suggestion
test: test-ci test-raw
```

File: infrastructure/scripts/check-license/Buildfile (hunk 1 above)
Comment (mnishamk-freiheit) [reply infrastructure/scripts/check-license/Buildfile:8]: Found that there is an issue in execution plan in which it wasn't captured. so created story to test it out and fix it.

need to test after this merges if the depends on is working properly. For the time being fixed it with adding redundancy

File: Makefile.variables (hunk 2 above)
Comment (mnishamk-freiheit) [reply Makefile.variables:20]: Yes, this will always be provided by humans.

File: infrastructure/scripts/check-license/Buildfile (hunk 1 above)
Comment (mnishamk-freiheit) [reply infrastructure/scripts/check-license/Buildfile:8]: [SRX-7B29XV](https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/-M37cZx7XYoOvQ5Mmcir/STORY/-N5_UFC3uri5Ovmf-wLv/)

File: charts/kuberpult/Makefile [hunk 4]
```diff
@@ -52,10 +52,14 @@ else
 	$(CT) lint --chart-yaml-schema=ci/chart_schema.yaml --lint-conf=ci/lintconf.yaml --target-branch=main --chart-dirs=. --charts=.
 endif
 
-test: ct-test $(TGZ_FILE)
+test-raw: $(TGZ_FILE)
```
Comment (sven-urbanski-freiheit-com) [inline charts/kuberpult/Makefile:55]:
```suggestion
test-helm: $(TGZ_FILE)
```

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: charts/kuberpult/Makefile (hunk 4 above)
Comment (mnishamk-freiheit) [reply charts/kuberpult/Makefile]: updated

**Pull request merged**